pkg/apis/application/v1alpha1/applicationset_types.go

```diff
@@ -917,33 +916,63 @@ func (a *ApplicationSet) RefreshRequired() bool {
 // If the applicationset has a pre-existing condition of a type that is not in the evaluated list,
 // it will be preserved. If the applicationset has a pre-existing condition of a type, status, reason that
 // is in the evaluated list, but not in the incoming conditions list, it will be removed.
-func (status *ApplicationSetStatus) SetConditions(conditions []ApplicationSetCondition, _ map[ApplicationSetConditionType]bool) {
-	applicationSetConditions := make([]ApplicationSetCondition, 0)
+func (status *ApplicationSetStatus) SetConditions(conditions []ApplicationSetCondition, evaluatedTypes map[ApplicationSetConditionType]bool) {
```

Comment:
Is the expectation that the evaluated types will be a 1:1 match of the types of the conditions in `conditions`? If so, why two vars to cary the same meaning?

Reply:
No, that is not how the function works as explained in the function description.